bindings/rust-examples/hyper-server-client/src/bin/server.rs

```diff
@@ -0,0 +1,78 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use clap::Parser;
+use http_body_util::{combinators::BoxBody, BodyExt};
+use hyper::body::Bytes;
+use hyper::service::service_fn;
+use hyper::{Request, Response};
+use hyper_util::rt::TokioExecutor;
+use std::error::Error;
+use tokio::net::TcpListener;
+
+/// NOTE: this certificate and key are to be used for demonstration purposes only!
+pub const CERT_PEM: &[u8] = include_bytes!(concat!(
+    env!("CARGO_MANIFEST_DIR"),
+    "/../certs/localhost-chain.pem"
+));
+pub const KEY_PEM: &[u8] = include_bytes!(concat!(
+    env!("CARGO_MANIFEST_DIR"),
+    "/../certs/localhost-key.pem"
+));
+
+#[derive(Parser)]
+struct Args {
+    #[clap(short, long, default_value = "localhost:1142")]
+    addr: String,
+}
+
+/// Echo the request body back to the client in the response.
+pub async fn echo(
+    req: Request<hyper::body::Incoming>,
+) -> Result<Response<BoxBody<Bytes, hyper::Error>>, hyper::Error> {
+    Ok(Response::new(req.into_body().boxed()))
+}
+
+async fn run_server(addr: &str) -> Result<(), Box<dyn Error>> {
+    // Configure the s2n-tls server.
+    let config = {
+        let mut builder = s2n_tls::config::Builder::new();
+        builder.load_pem(CERT_PEM, KEY_PEM)?;
+        // Enable HTTP/2 by including it in the server's supported ALPN values. The "http2"
+        // hyper-util feature must also be enabled.
+        builder.set_application_protocol_preference([b"h2"])?;
```

Comment:
I remember this was discussed in a previous PR. So we landed on manually setting the alpn via s2n-tls? Idk, this example might make me rethink that choice :/

Reply:
The ALPN doesn't need to be manually configured with s2n-tls-hyper. It will attempt to negotiate HTTP/2 by automatically setting the ALPN:
https://github.com/aws/s2n-tls/blob/3e57b05edc601351d3c5f2b03337d57de9129d5d/bindings/rust/standard/s2n-tls-hyper/src/connector.rs#L186-L200

This is a hyper server example which directly uses s2n-tls-tokio, not s2n-tls-hyper. s2n-tls-hyper is only used with the hyper client.